### ISSUE TITLE
Hello from Wyze and changes from wyze login APIs

### ISSUE THREAD

**Issue by xcz011** [mrlt8/docker-wyze-bridge#1163]
I’m an engineer at Wyze, and I wanted to inform you about some important updates to our Wyze Login API that may affect the docker-wyze-bridge project. As detailed in our recent forum post (Important Wyze App Information - 4/26/24), we will soon be shutting down the v2 and v3 APIs. This decision was made to enhance security and prevent credential stuffing attacks.

We encourage developers to transition to our supported Developer API for login processes. This change will ensure continued compatibility and enhance the security of integrations.

We appreciate the effort you have put into supporting Wyze and our ecosystem through your project. If you have any questions or need further details about the Developer API and how to transition, please feel free to reach out.

Thank you for your understanding and support.

**Comment by industrial64** [comment]
Can confirm, all Wyze cams and this project are now KIA. Time to ditch Wyze and buy RTSP (Reolink) cameras. 
Thanks for the heads up Wyze, but no native RTSP support is the end of you.

**Comment by letrain02** [comment]
> I’m an engineer at Wyze, and I wanted to inform you about some important updates to our Wyze Login API that may affect the docker-wyze-bridge project. As detailed in our recent forum post ([Important Wyze App Information - 4/26/24](https://forums.wyze.com/t/important-wyze-app-information-4-26-24/297488)), we will soon be shutting down the v2 and v3 APIs. This decision was made to enhance security and prevent credential stuffing attacks.
> 
> We encourage developers to transition to our supported Developer API for login processes. This change will ensure continued compatibility and enhance the security of integrations.
> 
> We appreciate the effort you have put into supporting Wyze and our ecosystem through your project. If you have any questions or need further details about the Developer API and how to transition, please feel free to reach out.
> 
> Thank you for your understanding and support.

If this makes wyze incompatible with this bridge I will be dumping wyze. Wyze has always been great. Sd card local storage. No requirements for paid monitoring. Security is important however wyze itself even had an rtsp firmware at one time. The ability to see stuff on my phone through an app is a perk. But local streaming to my own display is essential. As long as wyze plays nice with their API I don't see how this could be an issue. I'm more than happy to pay this developer for his services since wyze does not offer this ability itself.

**Comment by mr-bobdobalina** [comment]
> Can confirm, all Wyze cams and this project are now KIA. Time to ditch Wyze and buy RTSP (Reolink) cameras. Thanks for the heads up Wyze, but no native RTSP support is the end of you.

I've been testing TP-Link Tapo cameras. They're Wyze-like in that the cameras have local storage and/or cloud storage. They have RTSP built-in (needs to be enabled per camera and you lose cloud saving--no big deal to me) and they're cheap ($15.99 on Amazon right now for the C100 indoor model). I'd say video quality is on par with V3 Wyzecams. Their outdoor model (C120) is a little bulkier than Wyze but still fairly cheap at $29 and it is 2K video (I don't really notice a big difference from 1080p).

Video fed to my Blue Iris server has been SOLID. No drops or connection issues ever so far. Only downside I've found is the indoor models use an old school barrel connector power adapter. I've been thinking of getting a barrel->USB-A cable to see if that would work. The C120 does use USB-C however.

**Comment by xcz011** [comment]
This project already support newer developer login api and this change is only to retire legacy login api to enhance security. 

So this won't break the project in any way.

Thanks

**Comment by letrain02** [comment]
> This project already support newer developer login api and this change is only to retire legacy login api to enhance security. 
> 
> So this won't break this project in any way.
> 
> Thanks

I figured but you never know. Reddit screwed everyone with their API, and many companies do. 

I was a lucky one a couple days ago that decided to update which required a restart. Which then required API login. As long as wyze continues this way I will continue to place wyze cameras all over my 30 acres and recommend them to everyone I know.

Like I stated if wyze offered this feature I would use it. And honestly the fact the community is able to develop anything related to it is why wyze is so well used. 

 API login is easy to use in a docker. @mrlt8 would just have to change his documentation to state that it is required. I'd rather have an API exclusively than a password, but both are fine.

**Comment by anthonyjesmok** [comment]
I can confirm that the Wyze bridge is working once again with the new Developer API in place. Instructions are located at https://github.com/mrlt8/docker-wyze-bridge/wiki/Two-Factor-Authentication#wyze-api-keyid. 

Thank you to the Wyze team for engaging with this project and allowing the developer community to continue creating great additions for their awesome products!

**Comment by mr-bobdobalina** [comment]
> I can confirm that the Wyze bridge is working once again with the new Developer API in place. Instructions are located at https://github.com/mrlt8/docker-wyze-bridge/wiki/Two-Factor-Authentication#wyze-api-keyid.
> 
> Thank you to the Wyze team for engaging with this project and allowing the developer community to continue creating great additions for their awesome products!

Got mine working as well, thank you for the link!

**Comment by mike2246** [comment]
Working too, Thanks!

**Comment by mrlt8** [comment]
Thank you for the heads up @xcz011! I noticed the official app switched to oauth back in September so figured this would eventually happen. 

I'll update the documentation to indicate that the API key and ID are no longer optional but required now.

Several users have also noticed some issues connecting to cameras when on a separate network.  Can you confirm if the p2p and relay mode have been disabled on all 2024 firmware?

**Comment by cheme75** [comment]
Just curious if the keys/IDs are required if not using 2FA for wyze? 

I doubt I'll use my v2 cams at new house. I only have the original db setup right now. Anyway seems like a big hassle just for one db.

**Comment by mike2246** [comment]
Yes, I got api only. No 2fa

**Comment by cheesegrits** [comment]
@xcz011 thank you for reaching out.  It's good to see Wyze engaging with third party projects like this.

**Comment by vitosans** [comment]
> I’m an engineer at Wyze, and I wanted to inform you about some important updates to our Wyze Login API that may affect the docker-wyze-bridge project. As detailed in our recent forum post ([Important Wyze App Information - 4/26/24](https://forums.wyze.com/t/important-wyze-app-information-4-26-24/297488)), we will soon be shutting down the v2 and v3 APIs. This decision was made to enhance security and prevent credential stuffing attacks.
> 
> We encourage developers to transition to our supported Developer API for login processes. This change will ensure continued compatibility and enhance the security of integrations.
> 
> We appreciate the effort you have put into supporting Wyze and our ecosystem through your project. If you have any questions or need further details about the Developer API and how to transition, please feel free to reach out.
> 
> Thank you for your understanding and support.

Thanks for popping in and giving us the heads-up. RTSP might be out of your control, but it's appreciated that you help us navigate issues you might see.

**Comment by xcz011** [comment]
@mrlt8 I am not aware p2p relay change from my understanding, I will double check on that.
@cheme75 my understanding is  keys/ID will be required no matter of 2FA or not.

And appricated all of you to understanding of this. 
Feel free to close this issue whenever, Thanks again !!

**Comment by awilbourn** [comment]
My question is, why does the docker image still want the WZYE_EMAIL and WZYE_PASSWORD if we set the API_ID and API_KEY, seems like it would not be needed, but not able to get in without it.

**Comment by xcz011** [comment]
> My question is, why does the docker image still want the WZYE_EMAIL and WZYE_PASSWORD if we set the API_ID and API_KEY, seems like it would not be needed, but not able to get in without it.

@awilbourn https://support.wyze.com/hc/en-us/articles/16129834216731-Creating-an-API-Key wyze developer login api require to pass username, password, Keyid, Apikey. So docker image is correct

**Comment by mi5key** [comment]
> Can confirm, all Wyze cams and this project are now KIA. Time to ditch Wyze and buy RTSP (Reolink) cameras. Thanks for the heads up Wyze, but no native RTSP support is the end of you.

I bought Wyze cameras because I understood they were easy and effective because I moved into a house with no wired ethernet, so it was all WiFi.  I thought Wyze would help and be the solution.  I have 4 Reolink PoE cameras that are just waiting to be re-used.  I'm going to have to do that hard thing, and get PoE to my camera locations.

This project instilled hope, but it's clear Wyze really doesn't want to play in this space.  Ok then, bye.

**Comment by cheme75** [comment]
> > Can confirm, all Wyze cams and this project are now KIA. Time to ditch Wyze and buy RTSP (Reolink) cameras. Thanks for the heads up Wyze, but no native RTSP support is the end of you.
> 
> 
> 
> I bought Wyze cameras because I understood they were easy and effective because I moved into a house with no wired ethernet, so it was all WiFi.  I thought Wyze would help and be the solution.  I have 4 Reolink PoE cameras that are just waiting to be re-used.  I'm going to have to do that hard thing, and get PoE to my camera locations.
> 
> 
> 
> This project instilled hope, but it's clear Wyze really doesn't want to play in this space.  Ok then, bye.

I'm not a big fan of Wyze policies and practices since they shafted early adopters with the subscription money grab, but, if they didn't want to "play", they would not have provided this info or made it easy for us to continue to use these 3rd party solutions. I had 5 cams, but now down to just the db cam since I don't need much at new house. But the api key/ID was pretty darn simple to implement and my bridge works just fine.

**Comment by shawnsi** [comment]
@xcz011 late reply here, but wanted to check if this is in anyway related to problems users are experiencing with wyze api and local camera connections in the last 24 hours. There are several issues and discussions in this project, which are getting long in the tooth, but the TL;DR is that we can no longer connect to cameras unless they are on the exact same network subnet as our bridge software. This complicates many setups and simply isn't possible in some virtualization setups so far.

Anything you can share about planned wyze api changes that may be affecting this user community?

**Comment by java007md** [comment]
@xcz011  Can you provide any insights into the "IOTC_ER_TIMEOUT" errors that are impacting the community?

**Comment by xcz011** [comment]
@shawnsi  @java007md Thanks for messages. I have replied in this [thread](https://github.com/mrlt8/docker-wyze-bridge/issues/1438#issuecomment-2679316161). I will try to work with team and get back to community ASAP

**Comment by thachnb85** [comment]
None of my 7 cameras working with the latest API key/id from developer-api-console.wyze.com
I'm using version: 2.10.3 for Home Assistant OS.

**Comment by quochung9999** [comment]
> None of my 7 cameras working with the latest API key/id from developer-api-console.wyze.com
> I'm using version: 2.10.3 for Home Assistant OS.
> 
> 

Same here

**Comment by fabianosan** [comment]
Same here:

![Image](https://github.com/user-attachments/assets/3e4fa698-1e25-4761-b5c2-18a187044ce9)

**Comment by fabianosan** [comment]
This solved my problem: https://github.com/mrlt8/docker-wyze-bridge/issues/1432#issuecomment-2676301578

**Comment by BraveLittleRoaster** [comment]
> I’m an engineer at Wyze, and I wanted to inform you about some important updates to our Wyze Login API that may affect the docker-wyze-bridge project. As detailed in our recent forum post ([Important Wyze App Information - 4/26/24](https://forums.wyze.com/t/important-wyze-app-information-4-26-24/297488)), we will soon be shutting down the v2 and v3 APIs. This decision was made to enhance security and prevent credential stuffing attacks.
> 
> We encourage developers to transition to our supported Developer API for login processes. This change will ensure continued compatibility and enhance the security of integrations.
> 
> We appreciate the effort you have put into supporting Wyze and our ecosystem through your project. If you have any questions or need further details about the Developer API and how to transition, please feel free to reach out.
> 
> Thank you for your understanding and support.

Great, dude... do you HAVE ANY DOCUMENTATION BRO? WTF. FYI if anyone is having trouble reversing their f*ing API because they are as bad as me and DON'T PUBLISH THEIR DOCUMENTATION... just standard JA3 and JA4+ signature evasion will do the trick. Also block HSTS headers to avoid public key pinning on your emulator after install. JA3 evasion won't work if you're using BurpSuite so either make your own damn proxy or use a SOCKS5 based one to allow their built in f*king chromium to randomize its JA3 hash and evade the fking WAF since JA3 randomization won't work through anything but a socks5:// proxy. It really SHOULD NOT be this hard to access your API. I could write a f*king copyposta on how pissed off I am to get that working. I just wanna sync my Pan camera to my FACE bro! AR SECURITY GOGGLES BRO. Ugh, RELEASE THE APIIII DO IT. I had this working but updated past the supported firmware for this project. And now it doesn't work. Don't make me go to Ubiquiti I'LL f\*king do it bro. 

EDIT: Nvm just saw their prices. Come on, I'm poor we all are poor. Have you seen the damn grocery prices? Dude, we can only afford Wyze 😔 Let the people have the API!

<img width="168" height="300" alt="Image" src="https://github.com/user-attachments/assets/4c12afb7-c170-4bf1-a05f-99972aac7728" />

Edit: Edit: Sorry I was absolutely lit asf partying with friends. I'll leave it there. What I meant is you **need to bypass the JA3/JA4+ signature on their login** in order to actually log into the app without being redirected to "Getting Started" if you want to proxy with Burp or another tool like Postman in order to assist in fixing the Wyze community bridge. I'll either make my own golang middleware or fix this project eventually, as it broke the projects I was working on. To do this in BurpSuite you have to set TLS pass-through to all out-of-scope items, and set your scope to this: `(^|.*\.)wyze\.com$|(^|.*\.)wyzecam\.com$`

```
# TLS Pass-through domains
(^\|.*\.)ytimg\.com$
(^\|.*\.)youtube\.com$  
(^\|.*\.)rudderstack\.com$  
(^\|.*\.)rudderlabs\.com$  
(^\|.*\.)recaptcha\.net$  
(^\|.*\.)postscript\.io$  
(^\|.*\.)me\.gvt1\.com$ 
(^\|.*\.)livex\.ai$ 
(^\|.*\.)kaxsdc\.com$
(^\|.*\.)kaptcha\.com$
(^\|.*\.)gvt3\.com$
(^\|.*\.)gvt2\.com$
(^\|.*\.)gvt1\.com$
(^\|.*\.)gstatic\.com$  
(^\|.*\.)googleusercontent\.com$
(^\|.*\.)googletagmanager\.com$
(^\|.*\.)googleapis\.com$
(^\|.*\.)google\.com$
(^\|.*\.)google-analytics\.com$
(^\|.*\.)ggpht\.com$
(^\|.*\.)cloudflare\.com$
(^\|.*\.)braze\.com$
```

Note: After login you can set your TLS pass-through to just:

```
(^|.*\.)google\.com$
(^|.*\.)gstatic\.com$
(^|.*\.)googleapis\.com$
(^|.*\.)googleusercontent\.com$
(^|.*\.)recaptcha\.net$
(^|.*\.)kaptcha\.com$
```

Doing so will let you login to the app and then click around to generate the traffic you need to fix this. This is due to counter-bot fingerprinting on recaptcha detecting BurpSuite or Postman's JA3/4 signature typically. These kind of cloud WAFs actively block proxies, making fixing **undocumented APIs** a headache. You may also want to drop HSTS headers to avoid public key pinning. Burp has a match-and-replace rule for that baked in, so just check that off.

Another caveat is you need to use a rootable adb to sideload the Wyze APK. When building your image and selecting the API level, go to the "Services" drop down and choose "Google APIs". You will not be able to use `adb root` if "Google Play Store" is selected. Export your Burp cert as a .der and then run this script to install it into the system CA store. If you just use the UX to install a CA cert the normal way, it's installed in the user's CA store and that is ignored by the Wyze app for security reasons. Example: `./install_burp_cert.sh /path/to/BurpCert.der`

```shell
#!/usr/bin/env bash
set -euo pipefail

DER_CERT="${1:-}"
SERIAL="${2:-}"  # optional: emulator-5554, etc.

if [[ -z "$DER_CERT" || ! -f "$DER_CERT" ]]; then
  echo "Usage: $0 /path/to/burp-ca.der [adb-serial]"
  echo "Example: $0 ~/Downloads/burp_ca.der emulator-5554"
  exit 1
fi

if ! command -v openssl >/dev/null 2>&1; then
  echo "[-] openssl not found. Install it: sudo apt-get update && sudo apt-get install -y openssl"
  exit 1
fi

ADB="adb"
if [[ -n "$SERIAL" ]]; then
  ADB="adb -s $SERIAL"
fi

TMP_DIR="$(mktemp -d -t burpca.XXXXXX)"
trap 'rm -rf "$TMP_DIR"' EXIT

PEM_CERT="$TMP_DIR/burp-ca.pem"

echo "[*] Converting DER -> PEM..."
openssl x509 -inform DER -in "$DER_CERT" -out "$PEM_CERT"

echo "[*] Calculating Android CA hash (subject_hash_old)..."
HASH="$(openssl x509 -inform PEM -subject_hash_old -in "$PEM_CERT" | head -n 1)"
OUT_CERT="$TMP_DIR/${HASH}.0"
cp "$PEM_CERT" "$OUT_CERT"

echo "[*] Using hashed CA filename: ${HASH}.0"
echo

echo "[*] Waiting for device..."
$ADB wait-for-device

echo "[*] Restarting adbd as root (requires non-Google-Play / root-capable system image)..."
if ! $ADB root; then
  echo
  echo "[-] adb root failed."
  echo "    You are almost certainly using a Google Play (production) system image."
  echo "    Create a new AVD with an AOSP image (guaranteed root) or a Google APIs image (often rootable)."
  exit 1
fi

# Give adbd a moment to restart cleanly:
sleep 1
$ADB wait-for-device

echo "[*] Remounting /system as read-write..."
if ! $ADB remount; then
  echo
  echo "[-] adb remount failed."
  echo "    Make sure you started the emulator with: -writable-system"
  echo "    (Android docs: start with -writable-system, then adb remount.)"
  exit 1
fi

echo "[*] Pushing CA into /system/etc/security/cacerts/ ..."
$ADB push "$OUT_CERT" "/system/etc/security/cacerts/${HASH}.0"

echo "[*] Fixing permissions..."
$ADB shell chmod 644 "/system/etc/security/cacerts/${HASH}.0" || true
$ADB shell chown root:root "/system/etc/security/cacerts/${HASH}.0" || true

echo "[*] Rebooting emulator..."
$ADB reboot

echo
echo "[+] Done."
echo "    After boot: Settings -> Security -> Encryption & credentials -> Trusted credentials -> System"
echo "    You should see your Burp CA there."
```

I also made this helper script to sideload/install the `.apkm` downloaded off your favorite mirror. Example `/install_apkm.sh ~/Downloads/com.hualai_3.9.0.739-62439_1arch_7dpi_5lang_75471860e7b0e7563c96b55b422161ef_apkmirror.com.apkm`

```shell

#!/usr/bin/env bash
set -euo pipefail

APKM="${1:?Usage: $0 /path/to/app.apkm [serial]}"
SERIAL="${2:-}"

ADB="adb"
if [[ -n "$SERIAL" ]]; then
  ADB="adb -s $SERIAL"
fi

WORK="$(mktemp -d -t apkm.XXXXXX)"
cleanup() { rm -rf "$WORK"; }
trap cleanup EXIT

unzip -q "$APKM" -d "$WORK"

if [[ ! -f "$WORK/base.apk" ]]; then
  echo "ERROR: base.apk not found after extracting:"
  ls -la "$WORK"
  exit 1
fi

# Prefer abilist (can include arm64-v8a on images with ARM translation)
ABILIST="$($ADB shell getprop ro.product.cpu.abilist 2>/dev/null | tr -d '\r')"
if [[ -z "$ABILIST" ]]; then
  ABILIST="$($ADB shell getprop ro.product.cpu.abi | tr -d '\r')"
fi

IFS=',' read -r -a ABIS <<< "$ABILIST"

INSTALL=( "$WORK/base.apk" )

# Pick ONE matching arch split (if present)
FOUND_ARCH=""
for abi in "${ABIS[@]}"; do
  abi_tag="${abi//-/_}"  # arm64-v8a -> arm64_v8a
  f="$WORK/split_config.${abi_tag}.apk"
  if [[ -f "$f" ]]; then
    INSTALL+=( "$f" )
    FOUND_ARCH="$abi_tag"
    break
  fi
done

if [[ -z "$FOUND_ARCH" ]]; then
  echo "WARN: No matching arch split found for abilist=$ABILIST"
  echo "      (If this APKM is ARM-only and you're on AOSP x86_64, it will not install.)"
fi

# Add remaining non-arch splits (lang/dpi/etc), but skip other arch splits
ARCH_SPLIT_RE='split_config\.(arm64_v8a|armeabi_v7a|x86_64|x86)\.apk$'
shopt -s nullglob
for apk in "$WORK"/split_config.*.apk; do
  if [[ "$apk" =~ $ARCH_SPLIT_RE ]]; then
    continue
  fi
  INSTALL+=( "$apk" )
done
shopt -u nullglob

echo "Installing:"
printf '  %s\n' "${INSTALL[@]}"

$ADB install-multiple -r "${INSTALL[@]}"
echo "Success"


```

Once everything is ready you can proxy Wyze thru Burp via `adb shell settings put global http_proxy 10.0.2.2:8080`. If you want to disable the proxy for your emulator: `adb shell settings put global http_proxy :0`

**Comment by jdeath** [comment]
Posting here as this is pinned. Just found: `https://thingino.com/`

Replace firmware on Wyze Cam 1-3. Tried on my V3 and works great. Total replacement. Suggest switching if your camera is supported.
